### DISCUSSION TITLE
prod-friendly devtools (remove action menu)

### DISCUSSION THREAD

**Discussion by brantphoto** [TanStack/query#7696] [Ideas]
Is there any appetite for a production friendly version of devtools? (idea being to remove the action menu in the production version) There's some debate at my work about lazy-loading devtools and obscuring them (as mentioned in react query docs) but in the perfect world there might be a version of the tool that is more suited to production in that it can't set application state directly.

If this idea sounds ok, any idea if it's easy/hard/etc? Was thinking to submit a PR unless it's a rabbithole

<img width="917" alt="Screenshot 2024-07-08 at 3 47 34 PM" src="https://github.com/TanStack/query/assets/7061624/1ac71286-c01a-4206-a162-d9100a253bff">

**Comment by brantphoto** [comment]
can I put up a PR?

**Comment by DamianOsipiuk** [comment]
If you have access to the production app (valid credentials/token), then what prevents you from sending requests directly to your backend bypassing front-end entirely.

Whether devtools expose functionality to modify cache or not, you should not be able to modify `backend` state by changing front-end state.
The same way someone can open devtools and modify your front-end any way they want, or extract payloads and modify them they see fit.

Therefore i do not think we need to complicate devtools logic even more by preventing something that should not matter.
